src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -458,16 +458,35 @@ impl_display!(DbzMode);
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash, EnumKind)]
 #[enum_kind(ConnectorType)]
-pub enum CreateSourceConnector {
-    File {
-        path: String,
-        compression: Compression,
+pub enum CreateConnector<T: AstInfo> {
+    KafkaBroker {
+        broker: String,
+        with_options: Vec<SqlOption<T>>,
     },
-    Kafka {
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum KafkaSource {
+    Literal {
         broker: String,
         topic: String,
         key: Option<Vec<Ident>>,
     },
+    Reference {
+        connector: String,
+        topic: String,
+        key: Option<Vec<Ident>>,
+    },
+}
```

Comment:
This type seems like it wants to be split in two:

```suggestion
#[derive(Debug, Clone, PartialEq, Eq, Hash)]
pub enum KafkaSource {
    connector: KafkaSourceConnector,
    topic: String,
    key: Option<Vec<Ident>>,
}

#[derive(Debug, Clone, PartialEq, Eq, Hash)]
pub enum KafkaSourceConnector {
    Inline {
        broker: String,
    },
    Reference {
        connector: UnresolvedObjectName,
    },
}
```

Then you won't need to duplicate the parsing and display of `topic` and `key`.